src/query/catalog/src/database.rs

```diff
@@ -113,6 +113,20 @@ pub trait Database: DynClone + Sync + Send {
         Ok(())
     }
 
+    #[async_backtrace::framed]
+    async fn refresh_table(&self, _table_name: &str) -> Result<()> {
+        return Err(ErrorCode::TableEngineNotSupported(
+            "Only Iceberg catalog support refresh table cache",
```

Comment:
```
 Err(ErrorCode::TableEngineNotSupported(
            format!("Catalog {} don't support refresh database cache", self.name()),
        ))

or just return `Ok(())`
```

In the catalog trait, it does not know it has the iceberg implementation.
Don't broke the abstraction.

Reply:
Well, I also hesitated to go back directly. At present, it is a better choice to return unimpl and indicate the current catalog type?